libs/clients/syslumenn/src/lib/syslumennClient.utils.ts

```diff
@@ -301,3 +307,52 @@ export const mapEstateRegistrant = (
     bankStockOrShares: syslaData.bankareikningarVerdbrefEdaHlutabref ?? false,
   }
 }
+
+// TODO: get updated types into the client
+export const mapEstateInfo = (syslaData: DanarbuUppl): EstateInfo => {
+  return {
+    assets: syslaData.eignir
+      ? syslaData.eignir
+          .filter(
+            (a) =>
+              a.tegundAngalgs === TegundAndlags.NUMBER_0 &&
+              a?.tegundAngalgs &&
+              /^[fF]{0,1}\d{7}$/.test(a.fastanumer ?? ''),
+          )
+          .map(assetMapper)
+      : [],
+    vehicles: syslaData.eignir
+      ? syslaData.eignir
+          .filter((a) => a.tegundAngalgs === TegundAndlags.NUMBER_1)
+          .map(assetMapper)
+      : [],
+    ships: syslaData.eignir
+      ? syslaData.eignir
+          .filter((a) => a.tegundAngalgs === TegundAndlags.NUMBER_2)
+          .map(assetMapper)
+      : [],
+    cash: syslaData.eignir
+      ? syslaData.eignir
+          .filter((a) => a.tegundAngalgs === TegundAndlags.NUMBER_3)
+          .map(assetMapper)
+      : [],
+    flyers: syslaData.eignir
+      ? syslaData.eignir
+          .filter((a) => a.tegundAngalgs === TegundAndlags.NUMBER_4)
+          .map(assetMapper)
+      : [],
+    estateMembers: syslaData.erfingar
+      ? syslaData.erfingar.map(estateMemberMapper)
+      : [],
+    addressOfDeceased: syslaData?.logheimili ?? '',
+    caseNumber: syslaData?.malsnumer ?? '',
+    dateOfDeath: syslaData?.danardagur
+      ? new Date(syslaData.danardagur)
+      : new Date(),
+    districtCommissionerHasWill: Boolean(syslaData?.erfdaskra),
```

Comment:
I've noticed that in a few places theres a variable?.someField and sometimes there's variable.someField.
The question mark states that if the variable isn't there then undefined is returned so perhaps for consistency we could like in this case have syslaData?.field instead of sometimes having the question mark and sometimes not because if for whatever reason the syslaData isn't there then the lines that don't have the question mark will cause an error.

Reply:
This is difficult to analyze if a type was at one point `something | undefined` and later becomes `something`.
In this highlighted example I would've thought it was ok to do a check for existence and drop the guard after it is shown to exist.

Am I perhaps misunderstanding something.
Regardless, this raises a valid point and I'll do a pass over to find inconsistencies if possible. Thanks :)